src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ExclusionsHelper.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+internal static class ExclusionsHelper
```

Comment:
I'm remembering one of the main reasons I made this a static class... I wanted baselines that are used by multiple test cases to only be updated once per execution (the sdk content tests both use the SDK baseline, so that baseline should only be updated once). There are two issues to this:

1) Multiple tests can use multiple baselines, hence the nested dictionaries
2) xUnit tests don't share context

Clearly, I forgot about that second point when implementing this, since I thought that a static class would cover multiple tests. I think I'm going to follow https://xunit.net/docs/shared-context in order to accomplish this. Similarly, based on the xUnit documentation, I'm not convinced that I can have multiple instances of a fixture within a test class. If I can, there's not point in keeping the double nested dictionaries. Otherwise, I'm going to have to keep them and add locking for the Regex.

Reply:
Resolved by using a fixture for the sdk tests since these tests both use the assembly files. This has been implemented in https://github.com/dotnet/installer/pull/19243/commits/cdfcfd5099aeaf456bc78ab1afd62a4d31baf15c